### ISSUE TITLE
"Inspect" CLI commands should support JSON output

### ISSUE THREAD

**Issue by abrookins** [PrefectHQ/prefect#13188]
### First check

- [X] I added a descriptive title to this issue.
- [X] I used the GitHub search to find a similar request and didn't find it.
- [X] I searched the Prefect documentation for this feature.

### Prefect Version

2.x

### Describe the current behavior

Running a command like `prefect work-pool inspect <name>` always outputs object representations in Python objects.

### Describe the proposed behavior

Running a command like `prefect work-pool inspect <name>` should support a flag to output the result in JSON.

### Example Use

`prefect work-pool inspect <name> --output=json`
`prefect work-pool inspect <name> -o json`

### Additional context

There are times when it can be useful to save JSON representations to use as subsequent input to commands that expect JSON, such as `prefect work-pool update <name> --base-job-template <file.json>`.

**Comment by serinamarie** [comment]
Makes sense to me. I can implement a POC with the work I'm currently building in the global concurrency limit CLI.

**Comment by olanna12** [comment]
Hi @abrookins! I would like to be assigned this issue.

**Comment by trymzet** [comment]
IMO all CLI commands should support JSON output to allow scripting.

Currently, the issue is not only that output has to be parsed (so eg. a change in formatting of the output could break scripts), but that it can also be trimmed by Rich. So for semi-reliability, all scripts using Prefect CLI's output must currently set the `COLUMNS` env var to a high value (eg. 200) to ensure output is not trimmed.

**Comment by tsafacjo** [comment]
can I take it ?
